python-avd/schema_tools/avdschemaresolver.py

```diff
@@ -46,37 +51,38 @@ def _dynamic_keys(self, resolved_schema: dict):
 
             self.resolve(resolved_schema["dynamic_keys"][key])
 
-    def _items(self, resolved_schema: dict):
+    def _items(self, resolved_schema: dict) -> None:
         # Resolve the child schema
         # Repeat in case new refs inherited from the first ref.
         while "$ref" in resolved_schema["items"]:
             self._ref_on_child(resolved_schema["items"])
 
         self.resolve(resolved_schema["items"])
 
-    def _ref_on_child(self, resolved_schema: dict):
+    def _ref_on_child(self, resolved_schema: dict) -> None:
         """
-        This function resolves the $ref referenced schema,
-        then merges with any schema defined at the same level
+        This function resolves the $ref referenced schema, then merges with any schema defined at the same level.
 
         In place update of supplied resolved_schema
         """
         try:
             resolved = self.resolver.lookup(resolved_schema["$ref"])
         except PointerToNowhere:
+            msg = (
+                f"Unable to resolve $ref: '{resolved_schema['$ref']}'."
+                "Make sure to adhere to the strict format '^(eos_cli_config_gen|eos_designs)#(/[a-z$][a-z0-9_]*)*$'."
+            )
             raise RuntimeError(
-                (
-                    f"Unable to resolve $ref: '{resolved_schema['$ref']}'."
-                    "Make sure to adhere to the strict format '^(eos_cli_config_gen|eos_designs)#(/[a-z$][a-z0-9_]*)*$'."
-                )
+                msg,
             ) from None
         ref_schema = deepcopy(resolved.contents)
         resolved_schema.pop("$ref")
         merge(resolved_schema, ref_schema, same_key_strategy="use_existing", list_merge="replace")
 
-    def create_resolver(self, store: dict, base_uri=""):
+    def create_resolver(self, store: dict, base_uri: str = "") -> object:
```

Comment:
Can we not be more precise than `object`?

Reply:
No the actual type is intenal to the library, so we cannot import it.